controllers/argocd/deployment.go

```diff
@@ -626,19 +630,25 @@ func (r *ReconcileArgoCD) reconcileRedisDeployment(cr *argoprojv1a1.ArgoCD) erro
 // reconcileRedisHAProxyDeployment will ensure the Deployment resource is present for the Redis HA Proxy component.
 func (r *ReconcileArgoCD) reconcileRedisHAProxyDeployment(cr *argoprojv1a1.ArgoCD) error {
 	deploy := newDeploymentWithSuffix("redis-ha-haproxy", "redis", cr)
-	if argoutil.IsObjectFound(r.Client, cr.Namespace, deploy.Name, deploy) {
+
+	existing := newDeploymentWithSuffix("redis-ha-haproxy", "redis", cr)
```

Comment:
This line looks same as the above line ?

Reply:
yes, need _existing_ to check if any current change has happened so that the reconciler can update with the latest changes, in other deployments this comparison is in the end, so the assignment of these variables is not next to each other